lib/openscap_parser/profile.rb

```diff
@@ -20,8 +20,27 @@ def description
     end
 
     def selected_rule_ids
-      @selected_rule_ids ||= @parsed_xml.xpath("select[@selected='true']/@idref") &&
-        @parsed_xml.xpath("select[@selected='true']/@idref").map(&:text)
+      # Look for selected rule ids where the idref contains '_rule_' that is not preceded by 'group'
+      @selected_rule_ids ||= @parsed_xml.xpath("select[@selected='true']
+                                                      [contains(@idref, '_rule_')]
+                                                      [not(contains(substring-before(@idref, 'rule_'), '_group_'))]
+                                                      /@idref") &&
+                             @parsed_xml.xpath("select[@selected='true']
+                                                      [contains(@idref, '_rule_')]
+                                                      [not(contains(substring-before(@idref, 'rule_'), '_group_'))]
+                                                      /@idref").map(&:text)
```

Comment:
@skateman I tried using a lookahead/lookbehind regex solution and came up with something like this:
```
@parsed_xml.xpath("select[@selected='true']
                                 [matches(@idref, '^((?!_group_).)*?(_rule_).*$')]
                                 /@idref").map(&:text)
```
And I think it would've worked but `matches` is a function for XPath 2.0, not 1.0, so it's not available with Nokogiri. I thought you mentioned that there was a way to use Xpath 2.0 with Nokogiri but I tried looking it up and wasn't able to find anything... Most people just said try to make your solution work with the Xpath 1.0 functions so that's what I did here. 

This solution checks if the string contains `_rule_` and then if it does, it checks to make sure that the substring before `rule_` doesn't contain `_group_`.  Also, I'm checking the substring before `rule_` instead of `_rule_` in case there's an idref like `xccdf_org.ssgproject.content_group_rule_nfs_configuring_all_machines` because in this case, it would be a group, not a rule, so this will catch edge cases like this.

Reply:
According to their [docs](https://nokogiri.org/rdoc/Nokogiri/XML/Node.html#method-i-xpath) the `xpath` method can accept a custom handler class (last example) that you can use to implement your own matcher, which in our case can be something similar to the XPath 2.0 matches.

Reply:
The in-place class definition is a bit scary, but you can simply extract it into a separate file :wink: